### DISCUSSION TITLE
Upgrade with Cloud Object Store

### DISCUSSION THREAD

**Discussion by jgheagle** [Esri/arcgis-powershell-dsc#585] [Q&A]
Hi all - I'm wondering if there is a way to upgrade to 11.4 using the DSC which handles migration to a Cloud Object Store from Tile Cache Data Store. Or, if there is a way to register the cloud based object store for this purpose in a new install config? I'm not seeing this in the variable reference at the moment so my assumption is that these are manual steps at the moment. Is there anything on the agenda for this at the moment?

**Comment by cameronkroeker** [comment]
Hi @jgheagle,

A cloud object store can be registered with an ArcGIS Server 11.4 site using **ConfigData.Server.DataStoreItems** variable. Here is a sample:

https://github.com/Esri/arcgis-powershell-dsc/blob/539a5adada7de60e3b5ce801a74020c0e3112d61/SampleConfigs/v4/v4.4.0/Base%20Deployment/BaseDeployment-SingleMachine-CloudObjectStore.json#L112-L143

Thanks,
Cameron K.

**Reply by jgheagle** [reply]
Many thanks for this Cameron, quick question on this - is it possible to add these to the config on "upgrade" to register during upgrade? I tried this on a sandbox upgrade but it was ignored with not even an error in the log. Guessing it only works on a new install? If so, will still work this into our process for future installs.

**Reply by cameronkroeker** [reply]
Hi @jgheagle, that is correct the Cloud Object Data Store is only registered with `-Mode InstallLicenseConfigure`. After the upgrade completes, add the ConfigData.Server.DataStoreItems to the json config and re-run with `-Mode InstallLicenseConfigure`.
